### DISCUSSION TITLE
Speed boost with USB power modification.

### DISCUSSION THREAD

**Discussion by Stefankante** [outdoorbits/little-backup-box#201] [Show and tell]
I'm soooooo thrill to bring LBB on location set for file offloading and tried every possible way to make it as efficient as it could.
one thing poke me a lot is the speed when try to connect power hungry storage device like USB SSDs and card reader (Yes!!!!)

I found that Pi4 own USB ports can't supply sufficient power to those devices hence poor transfer speed. So I bypass 5V from Pi4's USBC power input to the USB3.0 5V pin. 

![IMG_3772 copy](https://user-images.githubusercontent.com/35141284/230230814-9bb4588f-e7a9-47d6-b983-3df16c5f544e.jpg)

Speed now sustained around high 90-110MB/s from something like 50-80MB/s and no dangling USB hub around as I tested both NVME USB and WD Passport HDD.
either with Sony CFexpress type A reader which also need considerably quite juice. 
All can powered by just decent powerbank.

Do at your own risk! But I'm here happy with my own box lol. 

![IMG_3781](https://user-images.githubusercontent.com/35141284/230230302-2b19deea-bc0f-41bb-a968-aa29a7e938e0.jpeg)
![IMG_3783](https://user-images.githubusercontent.com/35141284/230230276-4b0b80b5-ed12-43aa-9e75-9bd6d622a0e5.jpeg)
![IMG_3782](https://user-images.githubusercontent.com/35141284/230230297-d9cf5895-a206-476b-8d09-1e61863f5ec3.jpeg)

**Comment by outdoorbits** [comment]
Interesting hack! I linked your post in the [Wiki](https://github.com/outdoorbits/little-backup-box/wiki/02.-Hardware).

**Comment by Stefankante** [comment]
I used this powerbank DIY case with fast charge populated all holders except top one to put another powerbank circuit which small enough to place there and can supply plentiful 5V with enough current to run 2-3 HDD or card reader with Pi4 itself.

[http://th.aliexpress.com/item/1005004892808820.html?spm=a2g0o.detail.1000013.3.6644EG9SEG9Si3&gps-id=pcDetailBottomMoreThisSeller&scm=1007.13339.291025.0&scm_id=1007.13339.291025.0&scm-url=1007.13339.291025.0&pvid=a20d9767-0129-4432-9fd4-f025a3c6cec6&_t=gps-id:pcDetailBottomMoreThisSeller,scm-url:1007.13339.291025.0,pvid:a20d9767-0129-4432-9fd4-f025a3c6cec6,tpp_buckets:668%232846%238111%231996&pdp_npi=3%40dis%21THB%21259.32%21228.07%21%21%21%21%21%402103209b16812449499323559e60a8%2112000030918716189%21rec%21TH%21143148292](url)

[http://th.aliexpress.com/item/1005002350801045.html?spm=a2g0o.productlist.main.33.18fdNymRNymRm5&algo_pvid=8778716e-f787-443e-bf49-5259d0f3e3b4&algo_exp_id=8778716e-f787-443e-bf49-5259d0f3e3b4-16&pdp_npi=3%40dis%21THB%2186.39%2185.54%21%21%21%21%21%4021227f7e16812449344468276d071a%2112000020233242842%21sea%21TH%21143148292&curPageLogUid=SES5soZmefG9](url)

Why 2 chargers ? Because normally when you need to charge the powerbank it will restart and also charge only at 5V.

Pi4 internally connected to 2nd charger to prevent users using crap USB cable and decease chance of accidentally disconnected while copying.

![IMG_3940](https://user-images.githubusercontent.com/35141284/231282063-bbda788c-967f-4029-a640-485244d0fccc.jpg)

[http://th.aliexpress.com/item/1005004061734586.html?spm=a2g0o.productlist.main.111.285a5b0cqTfqe7&algo_pvid=deeee772-570d-42f4-a12d-2d760d6fa693&algo_exp_id=deeee772-570d-42f4-a12d-2d760d6fa693-55&pdp_npi=3%40dis%21THB%2168.69%2152.89%21%21%21%21%21%402102186a16812456577513128d06e6%2112000027906099335%21sea%21TH%21143148292&curPageLogUid=EOIeuoS7GArs](url)

Now use 2 chargers It could stay on while copying files and fast charge at the same time. The one use for charging also capable to supply USBC PD 12v without interrupting another charger which connect to pi4 all share same pool of batteries.

So I use USBC PD to 12v to power even large 3.5inch external drive.
Absolutely convenient possible by this project. Thanks!!!!!!

![IMG_3964](https://user-images.githubusercontent.com/35141284/231279924-051aba91-48df-4cd4-8f2e-be2c6fd988a6.jpeg)
![IMG_3968](https://user-images.githubusercontent.com/35141284/231279960-9c8b0d86-3c16-4611-9610-a697ac0fb78f.jpeg)

**Comment by mpwhiten** [comment]
I posted this separately but should have posted here instead... Here's my experience:

I have an RPi 4 model B running lbb and I wanted to prioritise backup speeds without carrying around an extra hub. I was running into issues with my setup where the backup would begin but after a few seconds it would freeze. I guessed this was due to insufficient power on the USB ports, as I am using both a source and target plugged into the 2x USB 3.0 ports. Since I do not want to carry an extra device (powered hub), I looked for alternative solutions.

I came across [this](https://raspberrypi.stackexchange.com/questions/104038/can-the-pi-4-power-2-external-usb-3-hdds) mod which wires the 5V GPIO line to the USB port, allowing for increased current to supply the ports. Sorry, [Stefankante](https://github.com/Stefankante), I didn't see your discussion first. I tried it myself and it's working great! The backups complete with no issues and I maintain around 80-90 MB/s throughout the transfer. Note that this will void your RPi warranty, so don't attempt if you don't want to lose your warranty.

I am using a [Sandisk Extreme PRO SD Card USB-C Reader](https://www.westerndigital.com/en-sg/products/accessories/sandisk-extreme-pro-uhs-ii-usb-3-0-type-c-sd-card-reader-writer?sku=SDDR-409-G46), which is great because I can plug it into my phone and it achieves excellent read speeds. I connect it to the RPi with a [Mcdodo OT-6550 ](https://www.mcdodo.com.ph/products/mcdodo-ot-6550-type-c-to-usb-a-3-0-converter) USB A-C adapter that supports USB 3.0 speeds. For the target storage device, I'm using a [Samsung DUO Plus 256GB USB Flash Drive](https://www.samsung.com/us/computing/memory-storage/usb-flash-drives/usb-3-1-flash-drive-duo-plus-256gb-muf-256db-am/) which supports high read and write speeds.

My next steps are to try different target devices. First a 2.5" SSD in a SATA - USB enclosure and then an NVMe M.2 SSD in a USB enclosure. I'll report back here with my findings.

Here's a photo of the mod on my RPi 4:

![RPi 4 5V USB mod](https://i.ibb.co/VCnJ3Xd/5V-mod.jpg)

**Comment by samuel-puschacher** [comment]
Hi,
I am using a Waveshare CM4 NAS Box

https://www.waveshare.com/cm4-nvme-nas-box.htm
I populted one of the PCIE slots with a NVME.
My speeds are as well 80-90MB/s, i think the difference between usb3 and usb3 vs usb3 and nvme is barely noticeable.
I could also share how I made the mounting for the nvme in oder that it shows up as /media/internal/
The second pcie slot I used an Adapter + AX210 Wifi card (you need to recompile the Kernel!) There I get arround 800mbit! with iperf3
Limiting factor here is the pcie2 x1 lane!
This Hardware therefore uses a pcie switch, so that ist the Bottleneck here.
![278590926-54c6c7bb-0eba-4b32-a943-615f2bf2121e](https://github.com/outdoorbits/little-backup-box/assets/9848082/b674330c-a43d-4b90-aba3-ea65c8b8c106)

It is also not getting better with PI5 as there are no Upgrades on pcie, just a Single PCIE line again.
So it probably will go towards different Hardware setup when more speed is needed, maybe LattePanda or OrangePi 5 plus ( PCIE 3 x4+ pcie 2 x1)

Addon for the AX210, I have issues when it is used in AP-Mode, then it starts crashing when a Client wants to connect. Some Posts suggest that these Intel Chips make issues in AP-Mode, so maybe some different Hardware would be better. In Client Mode it Works
My solution at the Moment was to file a hole for a third Antenne in the housing so I have 2x AX210 + 1x CM4 Wifi Antennas. And AP is always from internal Wifi, which can be achieved in /etc/comitup.conf with "primary_wifi_device: " Parameter is the interface name e.g. wlan0

**Comment by mpwhiten** [comment]
Thank you for that info @samuel-puschacher. I have now tested with an NVMe drive in an Orico M2PV-C3 enclosure (Realtek RTL9210 controller). I averaged around 50MB/s with this. I expect there is still either a power issue when using this NVMe drive or there is a compatibility issue, as it should be achieving higher speeds.

For now I am happy with the 80-90 MB/s I get with the Samsung flash drive and based on your testing this might be the maximum speed achievable due to limitations of the RPi.

**Comment by black-jmyntrn** [comment]
epic! thank you!
